Source/Plugin.BLE/Android/Device.cs

```diff
@@ -255,65 +256,75 @@ private Guid ParseDeviceId()
             return new Guid(deviceGuid);
         }
 
-        public static List<AdvertisementRecord> ParseScanRecord(byte[] scanRecord)
+        public static List<AdvertisementRecord> ParseScanRecord(byte[] scanRecord, BluetoothDevice nativeDevice)
```

Comment:
As mentioned above, the `nativeDevice` parameter is unnecessary, since the `Device` class has a `NativeDevice` member.

Do you want to fix the remaining comments yourself, or would prefer that I take over?

Reply:
I thought that passing the device through a parameter is a cleaner option, since it does not depend on the order of the function calls.
I hope I did everything right. But you can continue as you see fit.

Reply:
I actually missed that `ParseScanRecord` is declared as a static method, but simply removing the `static` (as you did) is certainly a good solution. Looks pretty good by now.